Cpm.py

```diff
@@ -4,43 +4,98 @@
 class Activity:
 
     def __init__(self, name, duration):
-        self._name = name
-        self._duration = duration
-        self._depending_activities = []
+        self.name = name
+        self.duration = duration
+        self.min_time_to_start = 0
+        self.max_time_to_start = 0
+        #self._depending_activities = []
 
     def __str__(self):
-        print("Activity %s with duration %s", self._name, self._duration)
+        print("Activity %s with duration %s", self.name, self.duration)
+
+    def __eq__(self, other):
+        return self.name == other.name and self.duration == other.duration
+
+    def __ne__(self, other):
+        return not self == other
 
 
 class Project:
 
     def __init__(self, project):
+        # this method initializes a project object if no dictionary or None is given, an empty dictionary will be used
         if project is not None:
-            self._project = project
+            self.__project = project
+        else:
+            self.__project = {}
+
+        self.__project_duration = 0
+
+    def add_depending_activity(self, activity, depending_activity):
+        # this method add depending activity to activity in the project
+        if activity in self.__project:
+            if depending_activity in self.__project[activity]:
+                self._project[activity].append(depending_activity)
+                logger.debug("%s added to the activity", depending_activity)
+            else:
+                logger.debug("%s already in the list of depending activities", depending_activity)
         else:
-            self._project = {}
-        self._project_duration = 0
+            logger.debug("%s isn`t at the project", activity)
 
-    def add_activity(self, name, duration):
-        self._project[name] = Activity(name, duration)
-        logger.debug("%s added to the project", self._project[name])
+    def add_activity(self, activity):
+        # this method add activity to the project
+        if activity not in self.__project:
+            self._project[activity] = []
+            logger.debug("%s added to the project", activity)
+        else:
+            logger.debug("%s already at the", activity)
 
     def remove_activity(self, activity):
-        if activity in self._project:
-            self._project.pop(activity)
+        # this method removing activity from the project
+        if activity in self.__project:
```

Comment:
lets log the `else` option: `logger.debug("%s trying to delete activity that not in the project", activity)`

Reply:
will modify tomorrow

Reply:
done